Correct import script

tricknotes opened the pull request:
HTML(original) -> Markdown -> HTML の変換の前後で、HTML 構造を壊していないかチェックしながら変換を行うようにしました。

from: https://github.com/bm-sms/daimon-news-multi-tenant/pull/226
## TODO
- [x] HTML への変換時に、連続する `<blockquote>` がひとつの要素になってしまうのでそれぞれ別の要素として変換されるようにする
- [x] 一旦モデルに各メソッドを定義してしまったので、これらを変換スクリプトに移動する

tricknotes committed:
Use nokogiri-diff

tricknotes committed:
Add `posts.original_html`

tricknotes committed:
Set `posts.original_html` on sync

tricknotes committed:
Extract models for WP

tricknotes committed:
Add import `post.original_html` script

tricknotes committed:
WIP

tricknotes committed:
save

tricknotes committed:
Compare HTML using pandoc

tricknotes committed:
Keep HTML comment

tricknotes committed:
Strip

tricknotes committed:
Remove unused

tricknotes committed:
Disable auto links in temporary

tricknotes committed:
Normalize text content

tricknotes committed:
Fix to normarize text content

tricknotes committed:
Normalize wrapped paragraph

tricknotes committed:
Format HTML diff

tricknotes committed:
Remove resolved TODO

tricknotes committed:
Remove unnecessary option

tricknotes committed:
Extract method

tricknotes committed:
Add linebreak to page separator in current HTML

tricknotes committed:
Add workaround for `****`

tricknotes committed:
Force convert `u` -> `string`

tricknotes committed:
Strip unnecessary tag

tricknotes committed:
Strip unnecessary attributes for `<img>`

okkez commented:
移行手順を考えながら、段階を踏んで移行できるように作業を進める。

okkez commented:
> HTML への変換時に、連続する <blockquote> がひとつの要素になってしまうのでそれぞれ別の要素として変換されるようにする

これは、間に `<!-- -->` のように空のコメントを挟むとよい。
Nokogiriで何か入れて、pandocで処理するときに消えないように工夫すればよい。
`&nbsp;` だと`<p>&nbsp;</p>`が入って段落が増えてしまうのでよくない。

tricknotes commented:
>> HTML への変換時に、連続する `<blockquote>` がひとつの要素になってしまうのでそれぞれ別の要素として変換されるようにする

> これは、間に `<!-- -->` のように空のコメントを挟むとよい。

別解として、`Redcarpet::Render::HTML` を拡張して blockquote の扱いを変更できたりしませんか？
(そのほうが、Markdown に変なハックをしなくて済むのなと思いました。)

あまり調べずにコメントしているので、もしそんなことはできないようなら空のコメントを挟むのでよいかと思います！

okkez commented:
syntax highlight 入れるときに renderer を拡張する必要があるので、そっちの方がいいかもしれませんねー。

okkez committed:
wip: Add HTML validator

okkez committed:
Remove needless require

okkez committed:
Invert condition

`validate` returns true when converted markdown is valid.

okkez committed:
Display diff when raise error

okkez committed:
Remove validator related methods from model

okkez committed:
Remove `_` prefix from private methods

okkez committed:
Use ATX header

okkez committed:
Extract WpHTMLUtil

okkez committed:
Cache markdown_text

okkez committed:
Use pandoc-ruby instead of reverse markdown

okkez committed:
Remove unused accessor

okkez committed:
Add TODO comment

okkez committed:
Use WpHTMLValidator

okkez committed:
Use proper name

okkez committed:
Add comment

okkez committed:
`convert_image_url` is invoked by `markdown_body`

okkez committed:
Render separated blockquote properly

okkez committed:
Add line breaks to close to the original behavior

okkez committed:
Move methods to WpHTMLUtil

okkez committed:
Add scripts/convert_html2md.rb

This script can convert original_html to markdown.  We use this script
when we change logic to convert original html to markdown.

okkez committed:
Reduce duplicate code

okkez committed:
Move current_html to WpHTMLUtil

okkez committed:
Add missing using

okkez committed:
Move `validate!` under Converter module

okkez committed:
Count errors

okkez committed:
Display HTML on error

okkez committed:
Indent

okkez committed:
Use `class_eval` instead of refinements

Because refinements does not work as I expected.

okkez committed:
Display markdown_body on error

okkez commented:
GFMが `<u>aaa</u>` サポートしてない。

okkez committed:
Use Daimon::Render::HTML

okkez committed:
Stop customizing `block_quote`

In previous version, many records are broken.
In this version(w/o customize), broken records are not so many.

okkez commented:
lib/daimon/render/html.rb を残してるのはシンタックスハイライトを見据えてのことです。

okkez committed:
Install pandoc using buildpack

okkez committed:
Parse command line options politely

okkez committed:
Rename

current_html ->
target_html

okkez committed:
Display sources

mtsmfm committed:
indent

mtsmfm committed:
require_dependency

mtsmfm committed:
image option is heroku-docker's option

mtsmfm committed:
rake db:migrate:reset

mtsmfm committed:
$LOAD_PATH includes /lib

mtsmfm committed:
require false

mtsmfm committed:
- -> /

mtsmfm committed:
typo

notmalize
normalize
  ^

mtsmfm committed:
post is nil

mtsmfm commented:
:shipit:

tricknotes commented:
Thanks for your careful work!!
